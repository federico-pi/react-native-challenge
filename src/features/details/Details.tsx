import React, { ReactElement } from 'react';
import { ImageBackground, Text, TouchableOpacity, View } from 'react-native';

import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
import LinearGradient from 'react-native-linear-gradient';

import { ChevronLeft } from '~assets/images';
import { ApiModels } from '~shared/models';
import { COLORS } from '~styles/defaults';
import { DetailsFeatureStyles } from '~styles/features';

const { DETAILS: styles } = DetailsFeatureStyles;

/**
 * The details screen component
 */
function Details(): ReactElement {
  const navigation = useNavigation();

  /**
   * Fetching params data
   */
  const route: RouteProp<
    {
      params: {
        item: ApiModels.SearchResponse;
      };
    },
    'params'
  > = useRoute();

  const { show } = route.params.item;

  console.log(JSON.stringify(show));

  return (
    <View style={[styles.container, styles.content]}>
      <TouchableOpacity
        style={styles.chevronContainer}
        onPress={() => navigation.goBack()}
      >
        <ChevronLeft stroke={COLORS.OPAQUE} />
      </TouchableOpacity>
      <ImageBackground
        style={styles.image}
        source={{ uri: show.image?.original }}
        resizeMode={'cover'}
      >
        <LinearGradient
          style={styles.gradient}
          colors={[
            /**
             * Specifying colors this way as a library requirement
             */
            'rgba(255, 255, 255, 0)',
            'rgba(255, 255, 255, 0.5)',
            'rgba(255, 255, 255, 1)',
          ]}
          // Defining the gradient brak points
          locations={[0.05, 0.4, 0.95]}
        />
      </ImageBackground>
      <View style={styles.info}>
        <Text numberOfLines={1} style={styles.title}>
          {show.name}
        </Text>
        <Text numberOfLines={3} style={styles.summary}>
          {show.summary
            .replace('<p>', '')
            .replace('</p>', '')
            .replace('<b>', '')
            .replace('</b>', '')}
        </Text>
      </View>
    </View>
  );
}

export { Details };
