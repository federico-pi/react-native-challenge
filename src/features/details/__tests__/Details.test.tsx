import React from 'react';

import { create, act } from 'react-test-renderer';

import { TvMazeMockData } from '~mocks';

import { Details } from '../Details';

const mockedNavigate = jest.fn();

const mockData = {
  show: TvMazeMockData,
};

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useRoute: () => ({
      params: {
        item: mockData,
      },
    }),
  };
});

test('snapshot', () => {
  let tree;

  act(() => {
    tree = create(<Details />);
  });

  expect(tree).toMatchSnapshot();
});
