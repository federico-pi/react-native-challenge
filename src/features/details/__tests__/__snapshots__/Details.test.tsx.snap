// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 54,
        "width": 40,
        "zIndex": 2,
      }
    }
  >
    <SvgMock
      stroke="#191D30"
      testID="back"
    />
  </View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 0.78,
        "justifyContent": "flex-end",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      colors={
        Array [
          "rgba(255, 255, 255, 0)",
          "rgba(255, 255, 255, 0.5)",
          "rgba(255, 255, 255, 1)",
        ]
      }
      locations={
        Array [
          0.05,
          0.4,
          0.95,
        ]
      }
      style={
        Object {
          "height": 60,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 0.22,
        },
        Object {
          "padding": 16,
          "paddingTop": 8,
        },
        false,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#191D30",
          "fontFamily": "Poppins-Bold",
          "fontSize": 24,
        }
      }
    >
      Under the Dome
    </Text>
    <Text
      numberOfLines={4}
      style={
        Object {
          "color": "#191D30",
          "fontFamily": "Poppins-Regular",
          "fontSize": 16,
          "marginTop": 4,
        }
      }
    >
      Under the Dome is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.
    </Text>
  </View>
</View>
`;
